Does not expose the interactive overrides in .NET Core

jmprieur opened the pull request:
Does not expose the interactive overrides in .NET Core of AcquireTokenAsync in .NET Core.

See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/557

jmprieur committed:
Does not expose the interactive overrides of AcquireTokenAsync in .NET Core.
See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/557

jmprieur commented:
@bgavrilMS @jennyf19 @henrik-me 
I thought we could do that part of the move to remove WinRT?

this will avoid complains from users that we expose something which is not implemented?

henrik-me commented:
I agree it's better to hide than to throw a NotSupportedException from somewhere. Aside: Will need to make the same changes to PublicClientApplication.

henrik-me reviewed (changes requested):
:clock1:

jmprieur committed:
Avoid exposing interactive overrides of AquireTokenAsync in .NET Core

This PR breaks the XForms project. Does it need to be .NET standard 1.1 ?

Severity Code Description Project File Line Suppression State
Error CS1061 'PublicClientApplication' does not contain a definition for 'AcquireTokenAsync' and no extension method 'AcquireTokenAsync' accepting a first argument of type 'PublicClientApplication' could be found (are you missing a using directive or an assembly reference?) XForms C:\gh\adal.net\msal\tests\dev apps\XForms\XForms\AcquirePage.xaml.cs 180 N/A
Error CS1061 'PublicClientApplication' does not contain a definition for 'AcquireTokenAsync' and no extension method 'AcquireTokenAsync' accepting a first argument of type 'PublicClientApplication' could be found (are you missing a using directive or an assembly reference?) XForms C:\gh\adal.net\msal\tests\dev apps\XForms\XForms\AcquirePage.xaml.cs 187 N/A

jmprieur commented:
dizzy me :) I started, was randomized, and when I got back thought I was done.
This PR breaks the msal\tests\dev apps\XForms\XForms\AcquirePage.xaml.cs as this is .NET standard 1.1

@jennyf19 @bgavrilMS does it need to be .NET standard 1.1 ? or can we do a bit better?
should we have a platform for .NET Core? (and which ?)

jennyf19 commented:
@jmprieur - I don't think the app needs to be 1.1 ...on another note... also need to remove the WebUIFactory from net1.3

---
In reply to: [410724671](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1170#issuecomment-410724671) [](ancestors = 410724671)

bgavrilMS commented:
@jmprieur - we discussed removing support for netstandard11 in MSAL but we do not have a good picture of other netstandard1.x libraries that take a depdency on it. Maybe it is a good time to get rid of it during the MSAL 2.0.0 preview and bring it back if needed?

jmprieur commented:
Yes, I was wondering if we should not expose the .NET Core platform (or platforms?). This would be unambiguous, and would enable us to not expose the interactive acquire token API in .NET Core (as this cannot be implemented).
Let's have a brainstorm on that.

jennyf19 commented:
I like this idea.

---
In reply to: [410734281](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1170#issuecomment-410734281) [](ancestors = 410734281)

henrik-me commented:
Are you proposing even removing netstandard13 ?

bgavrilMS commented on msal/src/Microsoft.Identity.Client/IPublicClientApplication.cs:
```diff
@@ -46,7 +46,7 @@ public interface IPublicClientApplication : IClientApplicationBase
         // expose the interactive API without UIParent only for platforms that 
         // do not need it to operate like desktop, UWP, iOS.
-#if !ANDROID
+#if !ANDROID && !FACADE
```
Looks like FACADE is only defined for netstandard1.1; The system automatically defines the compiler variables NETSANDARD1_1 and NETSTANDARD1_3 which are used here and there throughout the code. I think FACADE should be defined for both. Let me know if you want me to pick this up.

jennyf19 replied:
I agree FACADE should be defined for both. This would simplify things a lot.

bgavrilMS committed:
Disable AcquireTokenAsync in netstandard tfms

bgavrilMS commented:
I'm going to close this PR because the current approach will not work for code sharing through netstandard libraries.
